test/e2e/testsuites/dynamically_provisioned_pod_failover_with_replicas_tester.go

```diff
@@ -75,11 +75,23 @@ func (t *PodFailoverWithReplicas) Run(client clientset.Interface, namespace *v1.
 	}
 
 	//Check that AzVolumeAttachment resources were created correctly
-	allReplicasAttached := true
+	var allReplicasAttached bool
 	var failedReplicaAttachments *v1alpha1.AzVolumeAttachmentList
+
 	err := wait.Poll(15*time.Second, 10*time.Minute, func() (bool, error) {
+		failedReplicaAttachments = nil
+		allReplicasAttached = true
 		var err error
-		allReplicasAttached, failedReplicaAttachments, err = testtypes.VerifySuccessfulReplicaAzVolumeAttachments(t.Pod, t.AzDiskClient, t.StorageClassParameters, client, namespace)
+		var attached bool
+		var podFailedReplicaAttachments *v1alpha1.AzVolumeAttachmentList
+		for _, pod := range tDeployment.Pods {
+			attached, podFailedReplicaAttachments, err = testtypes.VerifySuccessfulReplicaAzVolumeAttachments(pod, t.AzDiskClient, t.StorageClassParameters, client, namespace)
+			allReplicasAttached = allReplicasAttached && attached
```

Comment:
`allReplicasAttached` defaults to `false` on line 78 above, so it can never be set to `true` in this expression.